src/concrete/ml/quantization/post_training.py

```diff
@@ -455,10 +456,12 @@ def _quantize_layers(self, *input_calibration_data: numpy.ndarray):
             has_variable_inputs = (len(curr_inputs) - len(curr_cst_inputs)) > 0
 
             variable_input_names = [
-                input_name for input_name in curr_inputs if input_name not in constants
+                input_name for input_name, _ in curr_inputs if input_name not in constants
             ]
             curr_calibration_data = tuple(
-                curr_inputs[input_name] for input_name in variable_input_names
+                input_data
+                for input_name, input_data in curr_inputs
+                if input_name in variable_input_names
             )
 
             # For mypy
```

Comment:
below this ( I can't comment further here) : 
- why casting is need : ` curr_calibration_data = cast(Tuple[numpy.ndarray], curr_calibration_data)`
- "# Find the unique integer producers of the current's op output tensor" what is a producer ? how are they used ?
-